### PR TITLE
Update to .NET 8.0 and refactor tests

### PR DESCRIPTION
**Pull request by TommyKolodziejczyk**
Updated target frameworks to .NET 8.0 in `launch.json`, `IntelOrca.OpenLauncher.Core.csproj`, `openlauncher.csproj`, and `IntelOrca.OpenLauncher.Tests.csproj`. v6 will be deprecated soon. v7 is already deprecated since May 14, 2024

 Upgraded various NuGet packages to their latest Minor versions. to not introduced regression i've upgraded only dotnet 8 files and few minor hotfix versions
 
 Refactored `BuildServiceTests.cs` to use `[Theory]` with `InlineData` and introduced a helper method for asset matching. -> to fix tests on local machine

### CODE REVIEW

File: src/IntelOrca.OpenLauncher.Core/IntelOrca.OpenLauncher.Core.csproj [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netstandard2.1</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (IntelOrca) [inline src/IntelOrca.OpenLauncher.Core/IntelOrca.OpenLauncher.Core.csproj:3]: Why was this changed?

Comment (TommyKolodziejczyk) [reply src/IntelOrca.OpenLauncher.Core/IntelOrca.OpenLauncher.Core.csproj:3]: dotnet standard is old way to get comaptibility with dotnetframework, dotnet 8 implements netstandard by default.

as state in below documentation -> .NET 5 and later versions .... eliminates the need for .NET Standard in most scenarios. 

https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#net-5-and-net-standard
[dotnet st](https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0)

going into details:
https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/

- Use netstandard2.0 to share code between .NET Framework and all other platforms.
- Use netstandard2.1 to share code between Mono, Xamarin, and .NET Core 3.x.
- Use net5.0 for code sharing moving forward.

in this use case luncher will be used only in net.5.0+ (currently net8) so this is not case to use netstandard and reason to move forward

_"While .NET Standard 2.1 will be the last version of .NET Standard, .NET 5 and all future versions will continue to support .NET Standard 2.1 and earlier. You should think of net5.0 (and future versions) as the foundation for sharing code moving forward. "_

Comment (IntelOrca) [reply src/IntelOrca.OpenLauncher.Core/IntelOrca.OpenLauncher.Core.csproj:3]: My reason for keeping it standard was in case we wanted to use it for a .NET framework app at some point. But I am happy to move it to .NET 8.

If you would like to keep this change, can you please fix the additional warnings that now get produced due to nullability checks being enabled by default.

Comment (TommyKolodziejczyk) [reply src/IntelOrca.OpenLauncher.Core/IntelOrca.OpenLauncher.Core.csproj:3]: updated :)
